### PR TITLE
SLCORE-434 Expose education principles in the More Info tab

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/rules/EducationPrinciples.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.rules;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.CheckForNull;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+
+class EducationPrinciples {
```
[the remark this answers is not in the document]
Comment (damien-urruty-sonarsource) [reply core/src/main/java/org/sonarsource/sonarlint/core/rules/EducationPrinciples.java]: bien vu !

File: core/pom.xml [hunk 2]
```diff
@@ -138,22 +138,6 @@
   </dependencies>
 
   <build>
-    <resources>
```
Comment (henryju) [inline core/pom.xml:141]: It was dead code right? Maybe should have been in a separate commit.

File: core/src/main/java/org/sonarsource/sonarlint/core/rules/CleanCodePrinciples.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import javax.annotation.CheckForNull;
+import org.apache.commons.io.IOUtils;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+
+class CleanCodePrinciples {
+
+  @CheckForNull
+  public static String getContent(String key) {
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/rules/CleanCodePrinciples.java:31]: @mithil-dobarkar-sonarsource in the end Damien moved content to resources. Maybe you can discuss with him the rational to see if you should do the same on your side for the "other" context content.

**Review by henryju — APPROVED** [review]
LGTM

File: core/pom.xml (hunk 2 above)
Comment (damien-urruty-sonarsource) [reply core/pom.xml:141]: Yep, split in 2 commits

File: core/src/main/java/org/sonarsource/sonarlint/core/rules/CleanCodePrinciples.java (hunk 3 above)
Comment (damien-urruty-sonarsource) [reply core/src/main/java/org/sonarsource/sonarlint/core/rules/CleanCodePrinciples.java:31]: My rationale is that it is static content, it was a bit ugly in the code (we don't have text blocks) and I had a lot of issues from Sonar :)
In this case I think it is more logical to have them as resources

**Pull request merged**